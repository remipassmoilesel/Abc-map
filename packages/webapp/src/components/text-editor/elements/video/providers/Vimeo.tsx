/**
 * Copyright © 2021 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

import React from 'react';
import { Provider } from '../Provider';
import { DefaultIntegration } from '../DefaultIntegration';

const urlRegexp = /vimeo\.com\/([0-9]+)/i;

export class Vimeo implements Provider {
  public canHandle(url: string): boolean {
    return !!this.extractVideoId(url);
  }

  public getIntegration(url: string): React.ReactNode {
    const videoId = this.extractVideoId(url);
    if (!videoId) {
      return <DefaultIntegration url={url} />;
    }

    return (
      <iframe
        title="Vimeo video player"
        src={`https://player.vimeo.com/video/${videoId}`}
        frameBorder="0"
        allow="fullscreen; picture-in-picture"
        allowFullScreen
      />
    );
  }

  private extractVideoId(url: string): string | undefined {
    const match = url.match(urlRegexp);
    if (!match || !match.length) {
      return;
    }

    return match[1];
  }
}
