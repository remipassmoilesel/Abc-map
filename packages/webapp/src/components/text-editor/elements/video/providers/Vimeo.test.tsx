/**
 * Copyright © 2021 Rémi Pace.
 * This file is part of Abc-Map.
 *
 * Abc-Map is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as
 * published by the Free Software Foundation, either version 3 of
 * the License, or (at your option) any later version.
 *
 * Abc-Map is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General
 * Public License along with Abc-Map. If not, see <https://www.gnu.org/licenses/>.
 */

import { render } from '@testing-library/react';
import { Vimeo } from './Vimeo';

describe('Vimeo', () => {
  let provider: Vimeo;
  beforeEach(() => {
    provider = new Vimeo();
  });

  it('should handle', () => {
    expect(provider.canHandle('https://vimeo.com/677973953')).toBe(true);
    expect(provider.canHandle('https://docs.slatejs.org/api/nodes')).toBe(false);
  });

  it('should return integration', () => {
    const { container } = render(<div>{provider.getIntegration('https://vimeo.com/677973953')}</div>);
    expect(container).toMatchSnapshot();
  });
});
