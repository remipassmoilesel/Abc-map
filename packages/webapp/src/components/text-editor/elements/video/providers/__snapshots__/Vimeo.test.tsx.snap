// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vimeo should return integration 1`] = `
<div>
  <div>
    <iframe
      allow="fullscreen; picture-in-picture"
      allowfullscreen=""
      frameborder="0"
      src="https://player.vimeo.com/video/677973953"
      title="Vimeo video player"
    />
  </div>
</div>
`;
